rundeckapp/grails-app/services/rundeck/services/ExecutionService.groovy

```diff
@@ -1976,6 +1990,16 @@ class ExecutionService implements ApplicationContextAware, StepExecutor, NodeSte
      */
     def Date parseRunAtTime(String runAtTime) {
         try {
+            return ISO_8601_DATE_FORMAT_XXX.get().parse(runAtTime)
+        } catch (ParseException e1) {
+
+        }
+        try {
+            return ISO_8601_DATE_FORMAT_WITH_MS_XXX.get().parse(runAtTime)
```

Comment:
Could be good to add a testcase for this format as well

Reply:
I will add testcases to "should scheduleAdHocJob with alternative ISO 8601 date" and "should not scheduleAdHocJob with invalid ISO 8601 date". 

But I don't know when this ISO_8601_DATE_FORMAT_WITH_MS_XXX format is used. Browsers are sending in ISO_8601_DATE_FORMAT_XXX.